File: server/src/main/java/io/microconfig/server/git/GitConfig.java
package io.microconfig.server.git;

import lombok.Data;
import lombok.extern.slf4j.Slf4j;
import org.eclipse.jgit.transport.CredentialsProvider;
import org.eclipse.jgit.transport.UsernamePasswordCredentialsProvider;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.stereotype.Component;

import java.io.File;

@Data
@Slf4j
@Component
@ConfigurationProperties(prefix = "git")
public class GitConfig {
    private File workingDir;
    private String remoteUrl;
    private String username = "open";
    private String password = "source";
    private String defaultBranch = "master";

    public CredentialsProvider credentialsProvider() {
        return new UsernamePasswordCredentialsProvider(username, password);
    }

    public String repoFolderName() {
        var slash = remoteUrl.lastIndexOf('/');
        var dot = remoteUrl.lastIndexOf('.');
        return remoteUrl.substring(slash, dot);
    }
}

File: server/src/main/kotlin/io/microconfig/server/git/GitConfig.kt
package io.microconfig.server.git

import org.eclipse.jgit.transport.CredentialsProvider
import org.eclipse.jgit.transport.UsernamePasswordCredentialsProvider
import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.boot.context.properties.ConstructorBinding
import java.io.File

@ConstructorBinding
@ConfigurationProperties(prefix = "git")
data class GitConfig(
    var workingDir: File,
    var remoteUrl: String,
    var username: String = "open",
    var password: String = "source",
    var defaultBranch: String = "master",
    var pullDelay: Long = 10
) {
    fun credentialsProvider(): CredentialsProvider {
        return UsernamePasswordCredentialsProvider(username, password)
    }

    fun dir(): File {
        return workingDir.resolve("repo")
    }
}